dependencies.yaml

```diff
@@ -143,8 +143,9 @@ dependencies:
         packages:
           - c-compiler
           - cxx-compiler
-          - nccl>=2.19
           - libucxx==0.42.*,>=0.0.0a0
+          - nccl>=2.19
+          - spdlog>=1.14.1,<1.15
```

Comment:
`spdlog` was missing in RAFT conda devcontainers. Previously it was gotten transitively from RMM but that was removed in https://github.com/rapidsai/rmm/pull/1739, causing a build failure. I discussed with @vyasr and we agreed this fix (adding `spdlog`) is necessary for the short term. In the medium-to-long term, @vyasr is going to rework the loggers.

Reply:
Thanks for summarizing that here, works for me.

(and thanks for alphabetizing this list while you're touching it).